flink-scala/src/main/scala/org/apache/flink/api/scala/codegen/TypeInformationGen.scala

```diff
@@ -144,21 +142,20 @@ private[flink] trait TypeInformationGen[C <: Context] {
           for (i <- 0 until getArity) {
             fieldSerializers(i) = types(i).createSerializer(executionConfig)
           }
+          // -------------------------------------------------------------------------------------
+          // NOTE:
+          // the following anonymous class is needed here, and should not be removed
+          // (although appears to be unused) since it is required for backwards compatibility
+          // with Flink versions pre 1.8, that were using Java deserialization.
+          // -------------------------------------------------------------------------------------
+          val unused = new SpecificCaseClassSerializer[T](getTypeClass(), fieldSerializers) {
 
-          new CaseClassSerializer[T](getTypeClass(), fieldSerializers) {
             override def createInstance(fields: Array[AnyRef]): T = {
```

Comment:
Why do we need to override this?

Reply:
we need to override it because the readObject of the super class is not being invoked.